instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt

```diff
@@ -336,3 +292,14 @@ class RouteRefreshTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.ja
             .build()
     }
 }
+
+private fun NavigationRoute.getDurationOfLeg(legIndex: Int): Double =
+    directionsRoute.legs()!![legIndex]
+        .annotation()
+        ?.duration()
+        ?.sum()!!
+
+private fun NavigationRoute.getIncidentsIdFromTheRoute(legIndex: Int): List<String>? =
+    directionsRoute.legs()?.get(legIndex)
+        ?.incidents()
+        ?.map { it.id() }
```

Comment:
in tests we must be good. // `nonEmpty` is optional
```suggestion
private fun NavigationRoute.getIncidentsIdFromTheRoute(legIndex: Int, nonEmpty: Boolean = true): List<String> =
    directionsRoute.legs()!!.get(legIndex)
        .incidents()
        .map { it.id() }
        .also { if(it.isEmpty && nonEmpty) {throws IllegalStateException() }
```

Reply:
`getIncidentsIdFromTheRoute` isn't supposed to verify something. It's just a quick way to access data we often need in the test above. If annotations will be null or empty the test will fail because the checks are:

```kotlin
assertEquals(
    listOf("11589180127444257", "14158569638505033").sorted(),
    refreshedRoutes[0].getIncidentsIdFromTheRoute(0)?.sorted()
)
assertEquals(
    listOf("11589180127444257"),
    initialRoutes[1].getIncidentsIdFromTheRoute(0)
)
assertEquals(
    listOf("11589180127444257", "14158569638505033").sorted(),
    refreshedRoutes[1].getIncidentsIdFromTheRoute(0)?.sorted()
)
```

Reply:
I mentioned that `// nonEmpty is optional`. The very first intention was to add `!!` to make tests stricter

Reply:
can you please share a case which current implementation won't handle?